src/pages/index.js

```diff
@@ -4,54 +4,149 @@ import Section from "../components/Section.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
 import "./index.css";
-import { config, initialCards } from "../utils/constants.js";
+import { config } from "../utils/constants.js";
+import { api } from "../components/Api.js";
+import Popup from "../components/Popup.js";
 
 const cardList = document.querySelector(".cards__list");
 
 const pictureModal = new PopupWithImage(".modal-picture");
 pictureModal.setEventListeners();
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      cardSection.addItem(createCard(cardData));
-    },
-  },
-  ".cards__list"
-);
-cardSection.renderItems();
+// const cardSection = new Section(
+//   {
+//     items: initialCards,
+//     renderer: (cardData) => {
+//       cardSection.addItem(createCard(cardData));
+//     },
+//   },
+//   ".cards__list"
+// );
+// cardSection.renderItems();
+
+const deleteModal = new PopupWithConfirm(".modal-delete");
+deleteModal.setEventListeners();
 
 const userInfo = new UserInfo({
   userNameSelector: ".profile__title",
-  userJobSelector: ".profile__description",
+  userAboutSelector: ".profile__description",
+  userAvatarSelector: ".profile__image",
 });
+
+let userId;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then((values) => {
+    userId = values[0]._id;
+    const userData = values[0];
+
+    const initialCards = values[1];
+
+    userInfo.setUserInfo({
+      name: userData.name,
+      about: userData.about,
+      avatar: userData.avatar,
+    });
+    const cardSection = new Section(
+      {
+        items: initialCards,
+        renderer: (cardData) => {
+          cardSection.addItem(createCard(cardData));
+        },
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+    return userData;
+  })
+  .catch((err) => {
+    console.error("Error. The request has failed: ", err);
+  });
+
 const profileModal = new PopupWithForm(".modal-profile", (inputs) => {
-  userInfo.setUserInfo(inputs);
-  profileModal.close();
+  profileModal.changeButtonText("Saving...");
+  api
+    .updateUserInfo(inputs)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        about: res.about,
+        avatar: res.avatar,
+      });
+    })
+    .catch((err) => {
+      console.error("Error. The request has failed: ", err);
+    })
+    .finally(() => {
+      profileModal.close();
+      profileModal.changeButtonText("Save");
+    });
 });
 profileModal.setEventListeners();
 
+const avatarModal = new PopupWithForm(".modal-avatar", (inputs) => {
+  avatarModal.changeButtonText("Saving...");
+
+  api
+    .updateProfilePicture(inputs.avatar)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        about: res.about,
+        avatar: res.avatar,
+      });
+    })
+    .catch((err) => {
+      console.error("Error. The request has failed: ", err);
+    })
+    .finally(() => {
+      avatarModal.close();
```

Comment:
`      avatarModal.close();
`

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well